### ISSUE TITLE
Attempting to create a route with a duplicate UUID overwrites existing data.

### ISSUE THREAD

**Issue by musoke** [OpenBeta/open-tacos#726]
Note: this was all in the test area; no real data was harmed.

## Steps to Reproduce

1. I copied 
```
8c1d30ab-260a-4e85-843c-bfc16455afec | A New Test Problem | V0+ | B
```
From [one test area ](https://openbeta.io/crag/3101f669-8b34-5f23-9a70-96f577b30d24) to [another](https://openbeta.io/crag/8539d051-b3b3-5d1c-8463-d2d188ed4756).  

2. Change grade to 5.10d and disciple code to S. 
```
8c1d30ab-260a-4e85-843c-bfc16455afec | A New Test Problem | 5.10d | S
```
4. Save.
5. See that route is added to page.
6. Refresh page.  See that route is not actually added.
![image](https://user-images.githubusercontent.com/16665084/224565848-88e9d93d-5ba6-4c30-bbb0-3f98fb785f66.png)

8. Check page from which it was copied, see that the discipline is updated.  The grade is still V-scale.
![image](https://user-images.githubusercontent.com/16665084/224565650-1be6a341-cd29-49ec-8e70-23f588acc7c9.png)

10. Check route's page.  See location, grade and discipline have changed.
![image](https://user-images.githubusercontent.com/16665084/224565728-501f9d88-f4e7-43be-815c-ffcfbbbb2b09.png)

## Expected Behavior

I expected that adding a route with a UUID would either move a route to a different page, add a second link to it, or display an error.  

## Current Behavior

The database becomes inconsistent.

The page initially shows the route as having been added, but upon refreshing it seems that a new route is not actually added to the area in which I saved it.

However, the existing route with that UUID is changed.
- the location is shown as the new area ([Dorian area](https://openbeta.io/crag/8539d051-b3b3-5d1c-8463-d2d188ed4756)) even though it doesn't appear on that page
- the discipline is sport
- the grade is 5.10d

It still appears on the original page,  [listed as sport](https://openbeta.io/climbs/8c1d30ab-260a-4e85-843c-bfc16455afec) on a [boulder](https://openbeta.io/crag/3101f669-8b34-5f23-9a70-96f577b30d24).  (I don't understand why, but far as I can tell areas with a mix of boulder problems and rope routes aren't allowed? This is strange to me but a separate matter.)

It is unclear exactly how many of these inconsistencies are due to stale caches.

## Possible Solution

A couple possibilities

- Check that UUIDs refer to routes already on a page
- Allow listing routes on more than one page

## Context (Environment)

I was checking if this is a viable way to move routes from one area to another.  Presumably a nicer way of doing this is planned.

**Comment by vnugent** [comment]
Thank you for reporting this bug. 

> I expected that adding a route with a UUID would either move a route to a different page, add a second link to it, or display an error.

Which behavior do you think is more logical?  
1. Move the route to the new area
2. Add the route as new (discard the copied UUID, generate a new UUID)
3. Error out?

**Comment by musoke** [comment]
I don't have a full understanding of the plan for the interface, so I'm not sure how my thoughts fit with your vision.

Is some version of this textbox planned as the long term interface for creating and reordering routes?

Option 1. would be nice for the short term; as far as I can tell there isn't yet a way to move routes between areas.  It might not make sense long term though, I'm guessing you have a plan for a nicer way to do that.  And a different method would still be needed to move areas and merge routes.

Either 2 or 3 seems sensible.   If option 2, is the one, it would be nice to display a warning that the UUID was discarded.

**Comment by vnugent** [comment]
Yes, I'd like to have a dedicate flow/UI to move a route (or an area) between other areas in the future.  For the time being I think we can go with option 2 & 3 (discard uuid, raise an error)

### Implementation ideas:

https://github.com/OpenBeta/open-tacos/blob/97680d95fccb83f40196af7071fe4f432b3b6567/src/components/editor/onChange.ts#L53

1.  Is first token a valid UUID?
  Yes, add code to compare this uuid with the default value (defaultClimbDict)

2. If uuid isn't the default value, return a new error string https://github.com/OpenBeta/open-tacos/blob/97680d95fccb83f40196af7071fe4f432b3b6567/src/components/editor/onChange.ts#L79

**Comment by snopan** [comment]
Hey @vnugent, I'm happy to pick this up.

I've had a read on the code and the problem, so what needs to be done is when `tokens[0]` is an `uuid`, we should check `defaultClimbDict` whether this `uuid` is an existing key in the dictionary. If it is then populate `errors` when it's returned. I have a question on this `errors` field, looking at the type it's a `Record` with only `gradeStr` and `disciplines` as error fields. Do I add `uuid` here to put uuid error message?

<img width="871" height="67" alt="Image" src="https://github.com/user-attachments/assets/e211579e-8eac-4012-a428-d3c98abbe48b" />


I'm still a bit new around this project so there's few questions I had
* How do we replicate this in the UI? I'm not sure how to move a climb and able to provide `uuid`?
* If the above is possible, is there a reason why we allow user to move a climb while providing the `uuid`, feels like this should be transparent to user.
* Looking at the code change it seems like this `uuid` check is enforced on the frontend, should this not be a backend check?
* I was reading `csvToClimb()` and it looked very complicated handling `tokens`. I was wondering if this can be simplified in any way? It looks like the `tokens` can be in the following format
  * uuid, name, gradeStr, disciplines
  * name, gradeStr, disciplines
Where `name`, `gradeStr`, `disciplines` are optional. If this is the case can the logic be, check if `tokens[0]` is `uuid` and handle accordingly. Then for the rest of the fields `name`, `gradeStr`, `disciplines` if they exist then we use them otherwise populate with default values. I feel like parts of the current logic works this way but it then does very specific length checks on the `tokens` array. I might be lacking context here so sorry if I'm over simplifying the problem if there's a specific reason why it's done this way.
